sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -23,7 +23,15 @@ class Span < OpenTelemetry::Trace::Span
 
         # The following readers are intended for the use of SpanProcessors and
         # should not be considered part of the public interface for instrumentation.
-        attr_reader :name, :status, :kind, :parent_span_id, :start_timestamp, :end_timestamp, :links, :resource, :instrumentation_library
+        attr_reader :name, :status, :kind, :parent_span_id, :start_timestamp, :end_timestamp, :links, :resource, :instrumentation_scope
+
+        # Returns an InstrumentationLibrary struct for backwards compatibility.
+        # @deprecated Please use instrumentation_scope instead.
+        #
+        # @return InstrumentationLibrary
+        def instrumentation_library
+          @instrumentation_library ||= InstrumentationLibrary.new(@instrumentation_scope.name, @instrumentation_scope.version)
+        end
```

Comment:
As with `SpanData`, I think callers should rely only on duck-typing.
```suggestion
        alias instrumentation_library instrumentation_scope
```